server/scsynth/SC_PortAudio.cpp

```diff
@@ -262,39 +273,168 @@ int SC_PortAudioDriver::PortAudioCallback(const void* input, void* output, unsig
     return paContinue;
 }
 
-void SC_PortAudioDriver::GetPaDeviceFromName(const char* device, int* mInOut) {
-    const PaDeviceInfo* pdi;
-    const PaHostApiInfo* apiInfo;
-    char devString[256];
-    PaDeviceIndex numDevices = Pa_GetDeviceCount();
-    mInOut[0] = paNoDevice;
-    mInOut[1] = paNoDevice;
+std::string SC_PortAudioDriver::GetDeviceName(int index) const {
+    auto* pdi = Pa_GetDeviceInfo(index);
+    std::string name;
+#ifndef __APPLE__
+    name += Pa_GetHostApiInfo(pdi->hostApi)->name;
+    name += " : ";
+#endif
+    name += pdi->name;
+    return name;
+}
+
+PaDeviceIndex SC_PortAudioDriver::GetPaDeviceFromName(const char* device, IOType ioType) const {
+    if (device == nullptr || device[0] == '\0')
+        return paNoDevice;
 
-    // This tries to find one or two devices that match the given name (substring)
-    // might cause problems for some names...
+    PaDeviceIndex numDevices = Pa_GetDeviceCount();
```

Comment:
(suggestion) minor nitpick -- `numDevices` is of type `PaDeviceIndex` but `i` in the loop below is of type `numDevices`. I realize they are probably the same thing or nearly the same thing, but it still looks sloppy (I know you didn't write this part)

Reply:
You mean `i` is of type `int`, right? This iteration was there before so I've just used that, but now I changed the type of `i` to `PaDeviceIndex` and it works.

Reply:
> You mean i is of type int, right?

yes, sorry, thanks for that